File: Submissions/1221_Split_a_String_in_Balanced_Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int result = 0;
        int balance = 0;
        for(auto& ch : s){
            if(ch == 'R') ++balance;
            else if(ch == 'L') --balance;
            if(balance == 0) ++result;
        }
        
        return result;
    }
};

File: Kotlin/1221_Split_a_String_in_Balanced_Strings.kt
class Solution {
    fun balancedStringSplit(s: String): Int {
        var result = 0
        var balance = 0;
        
        for(c in s){
            if(c == 'R') ++balance
            if(c == 'L') --balance
            if(balance == 0) ++result
        }
        
        return result
    }
}